pubtools/_quay/container_image_pusher.py

```diff
@@ -0,0 +1,259 @@
+import logging
+
+import requests
+
+from .exceptions import (
+    BadPushItem,
+    ManifestTypeError,
+)
+from .utils.misc import (
+    get_internal_container_repo_name,
+    log_step,
+)
+from .quay_api_client import QuayApiClient
+from .quay_client import QuayClient
+from .tag_images import tag_images
+from .manifest_list_merger import ManifestListMerger
+
+LOG = logging.getLogger("PubLogger")
+logging.basicConfig()
+LOG.setLevel(logging.INFO)
+
+
+class ContainerImagePusher:
+    """
+    Push container images to Quay.
+
+    No validation is performed, push items are expected to be correct.
+    """
+
+    def __init__(self, push_items, target_settings):
+        """
+        Initialize.
+
+        Args:
+            push_items ([_PushItem]):
+                List of push items.
+            target_name (str):
+                target name
+            target_settings (dict):
+                Target settings.
+        """
+        self.push_items = push_items
+        self.target_settings = target_settings
+
+        self.quay_host = self.target_settings.get("quay_host") or "quay.io"
+        if self.quay_host[-1] == "/":
```

Comment:
May I ask if there's a reason to not use `rstrip("/")`?

Reply:
There's no reason :) Fixed.